worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/FlushTask.scala

```diff
@@ -55,3 +56,31 @@ private[worker] class HdfsFlushTask(
     hdfsStream.close()
   }
 }
+
+private[worker] class S3FlushTask(
+    buffer: CompositeByteBuf,
+    val path: Path,
+    notifier: FlushNotifier,
+    keepBuffer: Boolean) extends FlushTask(buffer, notifier, keepBuffer) {
+  override def flush(): Unit = {
+    if (StorageManager.hadoopFs.exists(path)) {
+      val conf = StorageManager.hadoopFs.getConf
```

Comment:
Is there any difference for append between S3 and HDFS? What are the benefits of this logic?

Reply:
Yup. The main reason is S3 does't support append mode yet.

Reply:
@zhaohehuhu Is there another approach to support integrated a new `Filesystem` like s3 which cannot support `append`? Looks it is Inefficient, it copy the old data from s3 to worker and write to s3 again, this can scale out the write extremely.

Reply:
@zhaohehuhu That's great, look forward to know more about the new solution.

Reply:
Thanks @maobaolong